ls_formatting.go

```diff
@@ -60,6 +60,13 @@ func runLs(idLookup NameLookupFileLister, dirent os.FileInfo) string {
 		uid = lsFormatID(sys.UID)
 		gid = lsFormatID(sys.GID)
 	default:
+		if fiExt, ok := dirent.(FileInfoUidGid); ok {
```

Comment:
This is already in a type switch. 😉  You can just use:
```
case FileInfoUidGid:
```

Reply:
The type switch is on `dirent.Sys()` and not on `dirent`. That is usually the same of course.

The `.Sys()` function on a `fs.FileInfo` is something that does not really mix well with with interface assertions, I prefer to only use it if the fs.FileInfo originates from a "real filesystem operation".

Reply:
🤦‍♀️ But of course!